src/runtime/tests/storage-proxy-test.ts

```diff
@@ -807,4 +811,28 @@ describe('storage-proxy', () => {
     await engine.verifySubsequence('onHandleUpdate:P2:bar:+[v2](originator)');
     await engine.verify('HandleStore:bar:v2');
   });
+
+  it('ensures NoOpStorageProxy overrides all methods', async () => {
+    const engine = new TestEngine('arc-id');
+    const fooProxy = engine.newProxy({type: engine.type, name: 'foo'});
+
+    const properties = [];
+    let proto = Object.getPrototypeOf(fooProxy);
+    while (proto && proto !== Object.prototype) {
+      Object.getOwnPropertyNames(proto).forEach(name => {
+        const desc = Object.getOwnPropertyDescriptor(proto, name);
+        if (!!desc && typeof desc.value === 'function') {
```

Comment:
fyi, the `!!` isn't really necessary here; `desc` is already being treated as a "truthy" value due to the boolean context set up by the `&&` operator. `foo && foo.bar` is idiomatic JS.

Reply:
Got it, changed! :)